ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -503,7 +503,8 @@ public void maybeSetText(ReactTextUpdate reactTextUpdate) {
       // When we update text, we trigger onChangeText code that will
       // try to update state if the wrapper is available. Temporarily disable
       // to prevent an infinite loop.
-      getText().replace(0, length(), spannableStringBuilder);
+      getText().clear();
+      append(spannableStringBuilder);
     }
```

Comment:
Pull request https://github.com/safaiyeh/react-native/pull/2

```java
Integer currentPosition = getSelectionStart();
getText().clear();
append(spannableStringBuilder);
setSelection(currentPosition);
```

`setSelection` sets the cursor position to the previous one after append operation. 

This is some quick testing of the functionality with the react-native master branch. Let me know your feedback, I can publish a separate Pull Request if you are not available otherwise I can futher help you to review your pull request and test the changes.

| **BEFORE** | **AFTER** | 
|:-------------------------:|:-------------------------:|
| <img src="https://user-images.githubusercontent.com/24992535/83613179-b984ac00-a583-11ea-9b1c-3f0da6e4f5b5.gif" width="200" height="" /> | <img src="https://user-images.githubusercontent.com/24992535/83613371-f6e93980-a583-11ea-9fca-64f1a45f5c22.gif"  width="200" height="" /> |

Thanks a lot
Fabrizio Bertoglio

Reply:
@safaiyeh I'm preparing a pull request with this solution. Let me know if I should stop. I wish you a good weekend. Thanks. Fabrizio.

Reply:
https://github.com/facebook/react-native/pull/29070